ckanext/xloader/tests/test_jobs.py

```diff
@@ -1,740 +1,131 @@
-        assert job_dict["status"] == u"complete", job_dict
-        assert job_dict == {
-            u"metadata": {
-                u"ckan_url": u"http://www.ckan.org/",
-                u"datastore_contains_all_records_of_source_file": True,
-                u"datastore_active": True,
-                u"resource_id": u"foo-bar-42",
-            },
-            u"status": u"complete",
-        }
-
-        # Check the load
-        data = self.get_datastore_table()
-        assert data["headers"] == [
-            "_id",
-            "_full_text",
-            "date",
-            "temperature",
-            "place",
-        ]
-        assert data["header_dict"]["date"] == "TEXT"
-        # 'TIMESTAMP WITHOUT TIME ZONE')
-        assert data["header_dict"]["temperature"] == "TEXT"  # 'NUMERIC')
-        assert data["header_dict"]["place"] == "TEXT"  # 'TEXT')
-        assert data["num_rows"] == 6
-        assert data["rows"][0][2:] == (u"2011-01-01", u"1", u"Galway")
-        # (datetime.datetime(2011, 1, 1), 1, 'Galway'))
-
-        # Check it wanted to set the datastore_active=True
-        mocked_set_resource_metadata.assert_called_once()
-        assert mocked_set_resource_metadata.call_args[1]["update_dict"] == {
-            "datastore_contains_all_records_of_source_file": True,
-            "datastore_active": True,
-            "ckan_url": "http://www.ckan.org/",
-            "resource_id": "foo-bar-42",
-        }
-
-        logs = self.get_load_logs(job_id)
-        logs.assert_no_errors()
-
-        job = jobs_db.get_job(job_id)
-        assert job["status"] == u"complete"
-        assert job["error"] is None
-
-
-class Logs(list):
-    def get_errors(self):
-        return [message for level, message in self if level == "ERROR"]
-
-    def grep(self, text):
-        return [message for level, message in self if text in message]
-
-    def assert_no_errors(self):
-        errors = self.get_errors()
-        assert not errors, errors
-
-
-def get_sample_file(filename):
-    filepath = os.path.join(os.path.dirname(__file__), "samples", filename)
-    return open(filepath).read()
-
+from unittest import mock
 
+from ckanext.xloader import jobs
+from ckanext.xloader.utils import get_xloader_user_apitoken
+
+
+_TEST_FILE_CONTENT = "x, y\n1,2\n2,4\n3,6\n4,8\n5,10"
+
+
+def get_response(download_url, headers):
+    """Mock jobs.get_response() method."""
+    resp = Response()
+    resp.raw = io.BytesIO(_TEST_FILE_CONTENT.encode())
+    resp.headers = headers
+    return resp
+
+
+def get_large_response(download_url, headers):
+    """Mock jobs.get_response() method to fake a large file."""
+    resp = Response()
+    resp.raw = io.BytesIO(_TEST_FILE_CONTENT.encode())
+    resp.headers = {'content-length': 2000000000}
+    return resp
+
+
+@pytest.fixture
+def apikey():
+    try:
+        sysadmin = factories.SysadminWithToken()
+    except AttributeError:
+        # To provide support with CKAN 2.9
```

Comment:
Wouldn't `hasattr` be better, to avoid throwing and catching exceptions? Or even toolkit `check_ckan_version`

Reply:
Don't start [this holy war](https://docs.python.org/3.6/glossary.html#term-eafp) 🤣 

It's not a performance-critical fixture, so let's spare Patricio and leave the initial version here.

Reply:
I got curious: https://gist.github.com/pdelboca/bc5dc12a10718da5559be360437e280b

Reply:
Yeah, but it would be nice to mention that in your case numbers are approximately the same because of the complexity of `check_ckan_version`. I believe, for general cases when you are checking something plain, like `if 1 < 2:`, `if` is considerably faster. 
It doesn't change our situation, but it would be shame if someone finds your gist and use it to prove that `if` and `try` are the same all the time :)

Reply:
Eh, I hadn't realised that EAFP is an officially approved style. I don't like it myself, but okay.

Reply:
There isn't an approved style guide. I was just playing around because I got genuinely curious about it since there wasn't a clear definition of "better".

I refactored it to use `check_ckan_version`.